### PR TITLE
Added a cocktail system to replace cocktail reagents.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- Cuts out a lot of reagents and reactions purely used for cosmetic drinks.
- Removes a few /tg/ cocktails that are just references, or aren't cocktails (Sui Dream, Arak).
- Adds a cocktail system that changes the properties of the vessel based on the contents.
- This is prep for unifying materials and reagents as I really don't want to have to convert fifty or sixty fluff reagents when their main purpose is visual.
- Removes the Nothing reagent and associated drinks. Was discussed on Discord, it's a meme mime reagent.

**TODO**
- [x] Add ratio checking for cocktails instead of just presence.
- [x] Set ratios on cocktails.
- [x] Convert remaining cocktail reagents.
- [x] Remove primary reagent check, make the recipe tree commutative.
- [x] Remove remaining cocktail reagents.
- [x] Add codex page and adjust other codex page.
- [x] Readd and update cocktail descriptions.
- ~~Add reagent layering for cocktails.~~ This can be a future PR.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Crap, this was meant to be a draft.

File: code/modules/reagents/cocktails.dm [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+		return FALSE
+	var/list/check_ratios
+	var/i = 0
+	for(var/rtype in ratios)
+		i++
+		if(!prop.reagents.has_reagent(rtype) || (order_specific && prop.reagents.reagent_volumes[i] != rtype))
+			return FALSE
+		if(isnum(ratios[rtype]))
+			LAZYSET(check_ratios, rtype, ratios[rtype])
+	var/effective_volume = prop.reagents.total_volume - REAGENT_VOLUME(prop.reagents, /decl/reagent/drink/ice)
```
Comment (comma) [inline code/modules/reagents/cocktails.dm]: wot about cocktails where ice is an ingridient

Comment (MistakeNot4892) [reply code/modules/reagents/cocktails.dm]: There is exactly one of them now. Can duplicate the check for it though.

**Comment by MistakeNot4892** [comment]
Ready for review.

**Review by afterthought2 — APPROVED** [review]
I'm fine with this. The todo may need to be revisited.

**Pull request merged**